File: algo/python/number_of_divisors.py
# algo to find the number of divisors of a number n in O(sqrt(n)) time
def d_n(n):
    from math import sqrt
    count = 0
    for i in range(1, int(sqrt(n))+1):
        if n%i == 0:
            if i == n/i:
                count += 1
            else:
                count += 2
    return count

File: algo/cpp/number_of_divisors.cpp
// algo to find the number of divisors of a number n in O(sqrt(n)) time
int number_of_divisors(int n) {
    int c=0;
    for(int i=1;i<=sqrt(n); ++i) {
        if(n%i == 0){
                if(i == n/2)
                    c++;
                else
                    c+=2;
        };
    }
    return c;

}
